python/tests/test_training.py

```
@@ -0,0 +1,53 @@
+
+def test_sgd_training_loop():
+    """50 steps of SGD on fixed data. Loss at step 50 < loss at step 0."""
+    cfg = _make_config()
+    params = nl_hecate.init_params(cfg, 42)
+    input_ids, target_ids = _make_data(cfg)
+
+    initial_loss, _ = nl_hecate.forward(params, cfg, input_ids, target_ids)
+
+    lr = 0.01
+    for _ in range(50):
+        loss, grads = nl_hecate.compute_gradients(params, cfg, input_ids, target_ids)
```

Ruff (0.15.0)

[warning] Unpacked variable `loss` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)